primitives/io/src/lib.rs

```diff
@@ -1246,6 +1253,47 @@ pub trait Sandbox {
 	}
 }
 
+/// Runtime spawn extension.
+pub trait RuntimeSpawn : Send {
+	/// Create new runtime instance and use dynamic dispatch to invoke with specified payload.
+	///
+	/// Returns handle of the spawned task.
+	fn dyn_dispatch(&self, dispatcher_ref: u32, func: u32, payload: Vec<u8>) -> u64;
```

Comment:
I am not sure with `dyn_dispatch` as a name (I did read again the pr and got trouble locating where the wasm threads are spawned until I find this function implementation).
It seems more related to spawning a thread with a call (return type is the task id, 
Could think of `spawn_call`, `run`, `process`, `queue_process`. Having the first parameter called `dispatcher_ref` may be enough to indicate this will call an intermediate function to dynamically dispatch the call.